system/lib/standalone/standalone.c

```diff
@@ -195,33 +528,6 @@ void _wasmfs_get_preloaded_child_path(int index, void* buffer) {}
 
 void _wasmfs_get_preloaded_path_name(int index, void* buffer) {}
 
-// Import the VM's fd_write under a different name. Then we can interpose in
-// between it and WasmFS's fd_write. That is, libc calls fd_write, which WasmFS
-// implements. And WasmFS will forward actual writing to stdout/stderr to the
-// VM's fd_write. (This allows WasmFS to do work in the middle, for example, it
-// could support embedded files and other functionality.)
-__attribute__((import_module("wasi_snapshot_preview1"),
-               import_name("fd_write"))) __wasi_errno_t
-imported__wasi_fd_write(__wasi_fd_t fd,
-                        const __wasi_ciovec_t* iovs,
-                        size_t iovs_len,
-                        __wasi_size_t* nwritten);
-
-// Write a buffer + a newline.
-static void wasi_writeln(__wasi_fd_t fd, char* buffer) {
-  struct __wasi_ciovec_t iovs[2];
-  iovs[0].buf = (uint8_t*)buffer;
-  iovs[0].buf_len = strlen(buffer);
-  iovs[1].buf = (uint8_t*)"\n";
-  iovs[1].buf_len = 1;
-  __wasi_size_t nwritten;
-  imported__wasi_fd_write(fd, iovs, 2, &nwritten);
-}
-
-void _emscripten_out(char* text) { wasi_writeln(1, text); }
-
-void _emscripten_err(char* text) { wasi_writeln(2, text); }
-
```

Comment:
Why move this code up to the top?   Can we make the diff smaller by leaving it where it is?

Reply:
Because it's being called in methods above this (`__cxa_allocate_exception`/`__cxa_throw`/`_emscripten_throw_longjmp`), keeping it at the bottom will result in errors like `call to undeclared function '_emscripten_err'`.

Reply:
But `_emscripten_out` and `_emscripten_err` are declare in the `emscripten/console.h` header so that order here shouldn't matter.

Reply:
I had to move it because it didn't compile without it. Just tried to put it back and indeed doesn't work.
When I tried to include `emscripten/console.h` I got duplicate method definition errors. Could be my lack of C knowledge.

Reply:
I created a separate PR to address that issue: #18323.

Reply:
Nice, thanks!

Reply:
I merged in https://github.com/emscripten-core/emscripten/pull/18323 to fix the macro, it also needs the console error to be defined.